cmd/collections.go

```diff
@@ -0,0 +1,133 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"reflect"
+	"text/tabwriter"
+
+	"github.com/joinflux/webflowctl/internal"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	collectionsCmd.AddCommand(listCollectionsCmd)
+	collectionsCmd.AddCommand(getCollectionCmd)
+	rootCmd.AddCommand(collectionsCmd)
+}
+
+// collectionsCmd represents the webhooks command
+var collectionsCmd = &cobra.Command{
+	Use:   "collections",
+	Short: "Manage collections",
+	Long:  "List and manage collections",
+}
+
+type Collection struct {
+	Id           string `json:"_id"`
+	LastUpdated  string
+	CreatedOn    string
+	Name         string
+	Slug         string
+	SingularName string
+}
+
+// ListCollectionsResponse represents a response to the list collections request in Webflow.
+// See: https://developers.webflow.com/reference/list-collections
+type ListCollectionsResponse []Collection
+
+// listCollectionsCmd represents the command to list all collections for a site in Webflow.
+var listCollectionsCmd = &cobra.Command{
+	Use:   "list [site_id]",
+	Short: "list collections for a site",
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		siteId := args[0]
+
+		c := internal.NewClient(ApiToken)
+
+		body, err := c.Get([]string{"sites", siteId, "collections"})
+		if err != nil {
+			log.Fatalf("Request failed: %v", err)
+		}
+
+		var response ListCollectionsResponse
+		err = json.Unmarshal(body, &response)
+		if err != nil {
+			log.Fatalf("Failed to unmarshal response body: %v", err)
+		}
+
+		w := tabwriter.NewWriter(os.Stdout, 1, 1, 1, ' ', 0)
+		for _, collection := range response {
+			v := reflect.ValueOf(collection)
+			for i := 0; i < v.NumField(); i++ {
+				name := v.Type().Field(i).Name
+				value := v.Field(i).Interface()
+				fmt.Fprintf(w, "%s:\t%s\n", name, value)
+			}
+			fmt.Fprint(w, "\n")
+		}
+		w.Flush()
+	},
+}
+
+// GetCollectionResponse represents a response to the get collection request in Webflow.
+// See: https://developers.webflow.com/reference/get-collection
+type GetCollectionResponse struct {
+	*Collection
+	Fields []struct {
+		Id       string
```

Comment:
Does this need the `json:"_id"` annotation?

Related q: what's the purpose of these annotations?

Reply:
This doesn't need the annotation because the field is just called `id`. Go maps the struct property to the json field if they match but sometimes they don't (or can't like in this case) cause the field starts with `_id` and this can't be an exported struct property so the annotation in the other examples is mapping `struct.id` to the `_id` json property.